leakcanary-support-fragment/src/androidTest/java/com/squareup/leakcanary/tests/FragmentLeakTest.java

```diff
@@ -1,24 +1,37 @@
 package com.squareup.leakcanary.tests;
 
+import android.content.Intent;
+import android.support.test.InstrumentationRegistry;
+import android.support.test.espresso.Espresso;
+import android.support.test.espresso.matcher.ViewMatchers;
 import android.support.test.rule.ActivityTestRule;
 import android.support.v4.app.Fragment;
+import android.support.v4.app.FragmentActivity;
 import android.support.v4.app.FragmentManager;
+import android.widget.TextView;
+
 import com.squareup.leakcanary.InstrumentationLeakDetector;
 import com.squareup.leakcanary.InstrumentationLeakResults;
-import java.util.concurrent.CountDownLatch;
+import com.squareup.leakcanary.support.fragment.R;
+
 import org.junit.Rule;
 import org.junit.Test;
 
+import java.util.concurrent.CountDownLatch;
+
+import static android.support.test.espresso.action.ViewActions.click;
+
 public class FragmentLeakTest {
 
   @Rule
-  public ActivityTestRule<TestActivity> activityRule = new ActivityTestRule<>(
-      TestActivity.class);
+  public ActivityTestRule<FragmentActivity> activityRule =
+      new ActivityTestRule<>(FragmentActivity.class, true, false);
```

Comment:
Why did this need to change, and does that relate to adding `activityRule.launchActivity` in the test below?

Reply:
That's correct - it does. Instead of introducing a new `Test` class - I've updated the existing one to serve the needs of the new test case. With such setup, each test case in this class can start its own activity. This can be reverted if you prefer a separate test class for the new test.